### ISSUE TITLE
Error trying to start

### ISSUE THREAD

**Issue by scowling** [terrymacdonald/DisplayMagician#37]
**Describe the bug**
Installation seemed to go OK, however on trying to start the application I get the following error:

---------------------------
Fatal Error
---------------------------
The type initializer for 'DisplayMagicianShared.ProfileRepository' threw an exception.
---------------------------
OK   
---------------------------

**To Reproduce**
Well I installed it.. and tried to run it. To repro you'd need my exact setup :D

**Reporting Logs**
- Create a TRACE-level DisplayMagician.log file by running `C:\Program Files\DisplayMagician\DisplayMagician.exe --trace` in a command prompt.
_Trying to create this trace log results in the same error_

- The above command will create a DisplayMagician.log file in `C:\Users\<yourusername>\AppData\Local\DisplayMagician\Logs` which should contain a LOT of detailed TRACE level log entries. 
_However the log file got updated_

log shows one error
`2021-09-02 15:57:38.5595|ERROR|DisplayMagician.Program|Program/StartUpNormally top level exception: The type initializer for 'DisplayMagicianShared.ProfileRepository' threw an exception.:    at DisplayMagicianShared.ProfileRepository.get_AllProfiles()
   at DisplayMagician.UIForms.MainForm.RefreshNotifyIconMenus() in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\MainForm.cs:line 348
   at DisplayMagician.UIForms.MainForm..ctor(Form formToOpen) in H:\vscode-projects\DisplayMagician\DisplayMagician\UIForms\MainForm.cs:line 44
   at DisplayMagician.Program.StartUpApplication() in H:\vscode-projects\DisplayMagician\DisplayMagician\Program.cs:line 497 - System.ComponentModel.Win32Exception (0x80004005): The parameter is incorrect
   at WindowsDisplayAPI.DisplayConfig.PathDisplaySource.get_CurrentDPIScale()
   at DisplayMagicianShared.ProfileRepository.GenerateAllAvailableDisplayIdentifiers() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:line 1405
   at DisplayMagicianShared.ProfileRepository.IsPossibleRefresh() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:line 765
   at DisplayMagicianShared.ProfileRepository.LoadProfiles() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:line 676
   at DisplayMagicianShared.ProfileRepository..cctor() in H:\vscode-projects\DisplayMagician\DisplayMagicianShared\ProfileRepository.cs:line 82

`

**Enviroment (please complete the following information):**
 - Windows Version: Win10
 - DisplayMagician Version 1.1.4
 - Video Card: ASUS DUAL RX 5700 EVO OC
 - Number and make/model of monitors: 1x LG 34UC88-B 34in QHD Curved Ultrawide IPS LED Monitor, 1x Samsung Odyssey G5 QHD 144Hz FreeSync Curved 32in Monitor
 - Date of last video driver update: AMD Radeon 21.8.2 31 Aug 2021
 - Date of last windows update: 2021-09-01

**Comment by terrymacdonald** [comment]
Hi @scowling,

Can you try to get that TRACE DisplayMagician.log file again for me please? It contains a LOT of information that will help me troubleshoot the issue. Please note that the DisplayMagician.log file get overwritten each time you run DisplayMagician, so you will need to start DisplayMagician with the --trace option, and then close DisplayMagician (or have it crash), and then **straight away you need to copy the DisplayMagician.log file to somewhere else** so it won't get overwritten.

Looking at that error message on it's own it seems like a library that I use within DisplayMagician is having a problem when it's trying to understand your DPI settings. I'll have to see your TRACE DisplayMagician.log file to confirm this, but my gut feel is that WindowsDisplayAPI doesn't like your AMD video card :(.

But you're in luck! I'm currently doing a massive rewrite of DisplayMagician to add support for AMD Eyefinity spanned screens, and as part of that I'm writing my own video card libraries for NVIDIA, AMD and Windows that should stop this being an issue in the future. 

You're one of the only people I've communicated with that uses AMD Video Cards, so I'd LOVE it if you'd be able to do some testing for me! Is that something you'd be happy to do? The Windows and NVIDIA video card libraries are nearly finished, and I will finish off the AMD video card library once they're done. Will you have time in a couple of weeks to test out AMDInfo for me when I complete it? It will be available at https://github.com/terrymacdonald/AMDInfo/releases when I have something ready to go.

AMDInfo is my small little test program that will allow you to save and load AMD Eyefinity layouts and swap between them. It's the program I use for testing purposes so I can be sure that the video library is working before adding it into DisplayMagician. 

Can you help me add AMD Eyefinity support please?

Thanks
Terry

**Comment by scowling** [comment]
Sure! I'm a software developer so I understand the need to get field user
testing feedback.
Count me in.
Whole log file attached. Good luck.

**Comment by terrymacdonald** [comment]
Ah shoot. @scowling I can't see the log file attached anywhere within Github, or attached to the emailed version of the reply you send. Are you able to retry dropping the file into a reply directly within this Github issue? I know that process works ok: https://github.com/terrymacdonald/DisplayMagician/issues/37

I thought you might like to know the progress being made on DisplayMagician v2.0.0 too. It's has been a monster challenge, but I think I'm nearing the end of the NVIDIA chapter at least.  Things that the new DisplayMagician can do:

- Swap between surround and non-surround profiles within about 14 seconds
- Handle surround and non-surround screens together
- Works with NVIDIA mosaic cards now (custom AMD library part finished)
- Uses the PCI video card vendor ID to recognise NVIDIA and AMD vendors, and sets it's driver mode accordingly
- Can be 'forced' to use a different video card mode if needed (unlikely to be used, but hey, why not)
- Can set/clear a Desktop Background per profile if you want
- Stores and applies Windows HDR and NVIDIA HDR settings
- Records all settings provided by the NVIDIA driver and windows drivers under Windows 10, meaning that you display refresh settings are stored and set, as are bezel distances, or any other weird and wonderful settings you creative users are able to configure :)

But at present there are still bits I have to finish before I can craft any sort of production ready release:

- It is very slow to load. I've identified this is due to DisplayMagician keeping on calling the video card libraries. I need to implement some caching to ensure that the video detection is only run at programme launch and when screen layout changes in some way.
- I need to add AMD library support. This is the core reason why I started this journey, so it seems a little disingenuous to release DisplayMagician without this!
- I need to add a nice message that popups when an old DisplayMagician file is detected, that informs the user that this is a new version of DisplayMagician that uses a brand new DisplayProfile format that will require them to create a brand new set of Display Profile layouts.
- I need to test, and to provide my beta testers an early release of the software so that I iron out any bugs before a general release.
- I need to add the ability to test the upgrade mechanism without upgrading the production upgrade functionality. This is needed so I can iron out any issues before a general release. Very important as DisplayMagician now has over 1500 users!

And I think that's about it!

Some cool things to come for sure, and it's taken a lot of hard work to get this close. Another little while, and with your help @scowling, we're hopefully good to go!

Thanks
Terry

**Comment by scowling** [comment]
[DisplayMagician.log](https://github.com/terrymacdonald/DisplayMagician/files/7110875/DisplayMagician.log)

@terrymacdonald hopefully this way works.
Let me know when you have something for me to test
Cheers
Steve

**Comment by terrymacdonald** [comment]
Darn it. That DisplayMagician.log file unfortunately isn't a TRACE Level log file, as it doesn't contain any TRACE level log messages. It looks like it's the default INFO level log file. Unfortunately the INFO level log files don't give me enough information for me to know where to look.

I've looked at the error that was in that log file, and I really think it's an error in how the NvAPIWrapper library that DisplayMagician v1.14 reacts when it tries to access the currentDPIScale. TBH I don't think I'll be able to fix it for v1.1.4. I think that the original video card detection logic doesn't realise that your AMD card is not an NVIDIA card, as the log file says it's trying to initialise the NVIDIA NvAPIWrapper library. **It shouldn't be doing that.** It was supposed to realise it's not an NVIDIA card, and use the default windows video library.

Unfortunately there's not much I can do for v1.1.4 :(. The whole reason I'm creating DisplayMagician v2.0.0 is to sort out exactly these sorts of issues. And I'm making good progress. I just got NVIDIA display comparison logic to work today, and now I can swap between NVIDIA surround and NVIDIA non-surround profiles, as well as handle things like HDR, changed refresh rates and the like. 

It will probably be a few weeks before I'll be able to get the initial AMD config working. I had already started it previously, but had hit a snag that I needed AMD support to fix. So I'll revisit that once I get the final last few basic bugs ironed out of the software with the next few weeks testing.

I'll post in this issue thread once I need your help testing AMDInfo, and subsequently DisplayMagician too once I get the AMD library plumbed in.

Thanks
Terry

**Comment by scowling** [comment]
[DisplayMagician.log](https://github.com/terrymacdonald/DisplayMagician/files/7112420/DisplayMagician.log)

Sorry, full trace attached

**Comment by terrymacdonald** [comment]
I've just completed the first NVIDIA build of DisplayMagician, and I'll shortly be restarting the development of the AMD video manipulation library. The test version of DisplayMagician v2.0.0 currently won't work on any system that has an AMD video card in it. I've not finished the AMD library, yet I already plumbed AMD detection into DisplayMagician, which means that if you run DisplayMagician v2.0.0, it will attempt to use the non-existent AMD library if an AMD video card is detected, and it will just crash.

So, not so great for you at the moment, but I just wanted you to know I'm restarting the AMD stuff soon, and I hope to have something for you to test in a little while.  

I'll be getting you to test versions of AMDInfo in the near future until we get a fully working AMD video manipulation library, and then I'll port the library into AMD as the next step. My gut feel is that the process should take just over a month, but you never know with coding something as complicated as this!

Thanks for agreeing to help me out!
Terry

**Comment by scowling** [comment]
Thanks for the update. Good luck

**Comment by terrymacdonald** [comment]
Well now. It seems I'm faster than I thought I would be!

@scowling I now have the first working version of AMDInfo v0.0.5 partially working on my test system. It does enough to be able to save Eyefinity or non-eyefinity configurations, and then to apply them when loaded. That's just enough for us to test the logic and to make sure that the logic works on newer cards. I've only got an **ancient** AMD HD7790 that I've put in for testing purposes as it's the only AMD card I could get my hands on! I'm hoping that's good enough to iron out the majority of the problems :)

So, can you please test in the following way:

- Before anything, please download the latest AMDInfo ZIP file from https://github.com/terrymacdonald/AMDInfo/releases/tag/v0.0.5 and unzip it somewhere
- Firstly, set up your display configuration using AMD settings and the Windows Display settings exactly as you want to use them (e.g. one single AMD Eyefinity window using 3 screens)
- Next, save the settings you currently are using to a file to use later, using a command like `AMDInfo save triple-eyefinity-on.cfg`
- Next, change your display configuration using AMD settings and the Windows Display settings to another display configuration you'd like to have (e.g. 3 single screens without using AMD Surround)
- Next, save those settings to a different file to use later, using a command like `AMDInfo save triple-screen.cfg`

Now, try to swap between those two different settings using the load versions of the commands, e.g. `AMDInfo load triple-eyefinity-on.cfg` to load and apply the single eyefinity display profile, or `AMDInfo load triple-screen.cfg`

I successful run will say something like this:

```
AMDInfo v1.0.6
==============
By Terry MacDonald 2021

Attempting to apply AMD display config from plain-triple-screen.cfg
AMDInfo Display config successfully applied
```

Please let me know how that goes, and if you have any errors please upload your AMDInfo.log file here for me to test.

And thanks again for all your help so far. It's been a long standing goal of mine to add AMD support to DisplayMagician, so I really appreciate you helping me out!

Terry

**Comment by scowling** [comment]
Well done mate!

So I don't actually use eyefinity, but I set it up to test this. I have one
wide curved screen for developing, set up to be the most productive. Then I
have another curved wide screen on my rig. I currently just use a couple of
shortcuts linked to my start menu (eg. %windir%\System32\DisplaySwitch.exe
/external) which purely switches the output. (my rig used to be downstairs,
attached to my main PC via long HDMI and powered USB cables)

But for you I setup eyefinity and craned my neck around behind me to make
sure it worked .. and it did!

Attached is the AMDInfo.log file; let me know if you need anything else

Cheers
Steve

**Comment by terrymacdonald** [comment]
That's so excellent! Thanks for testing.

I've found a small error with non-eyefinity to non-eyefinity profile
swapping, so I'll fix that soon. I'll then probably ask if you can do a
couple more tests for me to check it works moving between three different
profiles types.

Thanks
Terry

**Comment by terrymacdonald** [comment]
OK! @scowling can you please test AMDInfo v1.0.0 for me? It's here: https://github.com/terrymacdonald/AMDInfo/releases/tag/v1.0.0

I did a LOT of testing over the last week, and I found quite a lot of edge cases. AMD's ADL video API is not very good. NVIDIA's is WAY easier to use. I found myself having to implement a few hundred lines of code just to check if AMD Eyefinity was enabled... NVIDIA does it in a single API call :(. But it's done now.

So, can you please test in the following way:

- Please download the latest AMDInfo ZIP file from https://github.com/terrymacdonald/AMDInfo/releases/tag/v1.0.0 and unzip it somewhere
- Firstly, set up your display configuration to combine your two screens into a single AMD Eyefinity screen
- Next, save the settings you currently are using to a file to use later, using a command like `AMDInfo save eyefinity-on.cfg`
- Next, change your display configuration using AMD settings and the Windows Display settings to turn off the Eyefinity, and to set your main screen (the one in front of you) as the main window,
- Next, save those settings to a different file to use later, using a command like `AMDInfo save dual-screen-front-main.cfg`
- Next, change your display configuration using Windows Display settings to set your side screen (the one in behind you) as the main window,
- Next, save those settings to a different file to use later, using a command like `AMDInfo save dual-screen-back-main.cfg`

Now, try to swap between those three different settings using the load versions of the commands, e.g. `AMDInfo load eyefinity-on.cfg` to load and apply the single eyefinity display profile, or `AMDInfo load dual-screen-front-main.cfg` or `AMDInfo load dual-screen-back-main.cfg`

My goal here is to make sure that you can swap freely between all the different config files without any issues. If you can do that without problems, then I can continue working integrating the AMD library into DisplayMagician, and just concentrate on getting DisplayMagician v2.0.0 working.

Thanks for all your help so far!

Terry

I successful run will say something like this:

**Comment by scowling** [comment]
Switching from main to back works, switching to eyefinity errors
Config and log file attached

**Comment by terrymacdonald** [comment]
Oh I forgot to mention that you'll need to delete the old CFG file, and
then create new ones again. I changed the CFG file structure quite
drastically, and so the old files won't work.

Did you reuse the old cfg files in the tests you just did with the new
AMDInfo v1.0.0? Or was that with new files created now?

Terry

**Comment by scowling** [comment]
Pretty sure they were new ones, unless I happened to choose the same file
name...

**Comment by terrymacdonald** [comment]
Darn.  Ok I'll have a look tomorrow and see what I can see. Thanks for
testing so quickly!

**Comment by scowling** [comment]
I extracted it into its own folder, so definitely different configuration
files.
No problem. Watching our football finals and having a quiet beverage. Happy
to help 😁

**Comment by terrymacdonald** [comment]
Oh, one thing I forgot to ask. Can you please upload the 3 CFG files you created. It will give me a great way to compare what AMDInfo made for you against what AMDInfo made for me. There may be a difference in how our video cards react, so it may help me avoid the error on your system with the next version.

Thanks. Good to hear you're getting some relaxing done. There's not been too much opportunity for that over the last couple of years!

Terry

**Comment by scowling** [comment]
Yeah I uploaded them with the earlier reply 👍

**Comment by terrymacdonald** [comment]
Oops! Sorry about missing them. It's quite late here and I obviously need
some sleep 😁

**Comment by terrymacdonald** [comment]
Hi @scowling,

Unfortunately it seems that replying to Github via email doesn't attach any files sent along with thee reply. It also didn't pass them on to me via email either :(.  Any chance you can visit the issue at https://github.com/terrymacdonald/DisplayMagician/issues/37 and attach the AMDInfo.log and the 3 config files again?

Sorry about that, and thanks again!

Terry

**Comment by scowling** [comment]
[AMDInfo.v1.0.0-logs-and-config.zip](https://github.com/terrymacdonald/DisplayMagician/files/7192970/AMDInfo.v1.0.0-logs-and-config.zip)
Here you go @terrymacdonald

**Comment by terrymacdonald** [comment]
Thanks  Steve,

Are you able to attempt to load .\eyefinity-on.cfg and send me the logs of
that? The AMDInfo.log file you've sent appears to be a successful change
back to dual-scree-fron-main.cfg, but I need the AMDInfo.log file from a
failed eyefinity-on.cfg load. Really sorry to ask you to do yet more work
again, but you're the only AMD User I know!

Also, as another little tidbit of information, I just managed to get the
AMD library integrated and running into DisplayMagician v2.0.0 last night!
I'm still 2-3 weeks from releasing another alpha, but if we can crack the
AMD Eyefinity problems together then you'll be able to use the next release
of DisplayMagician and all its new features.

So thank you again for all your help.

Thanks

*Terry MacDonald *| Principal Security Consultant | Chief Operating Officer



M: +64 211 918 814 <+64+211+918+814>
E: ***@***.***
W: www.cosive.com

**Comment by scowling** [comment]
[AMDInfo.log](https://github.com/terrymacdonald/DisplayMagician/files/7193124/AMDInfo.log)
Here you go @terrymacdonald - Happy to help

That's great news re/ the integration.

**Comment by terrymacdonald** [comment]
Hi @scowling,

So this line tells me that the library fails when it's trying to turn on the Eyefinity profile that was saved earlier:

`2021-09-20 09:31:32.5234|ERROR|DisplayMagicianShared.SharedLogger|AMDLibrary/SetActiveConfig: ERROR - ADL2_Display_SLSMapConfig_SetState returned ADL_STATUS ADL_ERR when trying to set the SLSMAP with index 0 to TRUE for adapter 0.`

There must have been an AMD Eyefinity/SLS profile setup within AMD Radeon Setup when AMDInfo took the settings, but it says that AMD SLS Profile no longer exists. This can happen if the AMD Profile was removed, which is REALLY easy to do with the poor UI that AMD have made for it. I'll need you to do a specific series of steps to confirm whether this is the case or not.

Can you please open up AMD Radeon Setup, click on the Gear icon and click on Display sub-menu. Then go to the Eyefinity section. Does the AMD Eyefinity option read as 'Quick Setup' or 'Discard Setup'? If it's 'Quick Setup', then that means there isn't an existing SLSMap stored by the driver to use at present. This would be the cause of our error. AMDInfo assumes that the Eyefinity is always stored by the AMD Radeon software, and we just turn it on when we find it. The problem is that we didn't find it :(.

If the AMD Eyefinity option reads as 'Quick Setup', can you please perform these steps in order:
- Set up a new AMD Eyefinity Profile
- Run `AMDInfo save new-eyefinity-on.cfg` to save the new config
- Close AMD Radeon Setup window, leaving Eyefinity on.
- Run `AMDInfo load dual-scree-fron-main.cfg` to swap back to your two single screens
- Check that it has swapped properly (and please save the AMDInfo.log into a different file name so you can send it to me)
- Run `AMDInfo load new-eyefinity-on.cfg` to attempt to turn the Eyefinity configuration back on.
- Check that it has swapped properly into Eyefinity mode (and please save the AMDInfo.log into a different file name so you can send it to me)

Then please reply to this message with the results of your testing, and the two different AMDInfo.log files for the two different runs.

My gut feel is that I will have to add the ability for the AMDInfo program to try to create a brand new Eyefinity session if it can't find one there already, which is quite a bit more work than I was hoping to do, but I might have to do this to ensure robustness of the system, and to avoid overly specific instructions for users. I always prefer to make my job difficult and make the users job easy, and I think this is an instance of this :).

I *should* really do the same for the NVIDIALibrary as well, just in case NVIDIA decide to change their NVIDIA Setup UI like AMD did, as this will just make everything much more likely to work reliably. Le sigh. More things to do :).

Thanks
Terry

**Comment by scowling** [comment]
100% dodgy UI. I had just discarded the Eyefinity setup to get back to normal.
All working fine now, step 1 and 2 log files attached

[AMDInfo-step2.log](https://github.com/terrymacdonald/DisplayMagician/files/7193521/AMDInfo-step2.log)
[AMDInfo-step1.log](https://github.com/terrymacdonald/DisplayMagician/files/7193522/AMDInfo-step1.log)

**Comment by terrymacdonald** [comment]
Awesome! This was 100% useful to me though, as I now know what software dev I need to do to make this all a bit easier to use. My plan is to have it create a new SLS config within AMD driver land so it won't matter what their UI does. It will definitely reduce my support calls ;D.

Thank you so much for your testing. You have literally saved my future self from a wall of support issues!

I don't think I'll need your help testing AMDInfo now, but I will ping you again to test the next DisplayMagician v2.0.0 alpha version once I get these additional tweaks added in.  It's getting closer!

Terry

**Comment by scowling** [comment]
All good. Great work so far.

**Comment by terrymacdonald** [comment]
OK! One final test I think I need from you to confirm the AMD stuff is working. I have to shortly change out my old AMD video card to do some more NVIDIA video card testing, so I'd really like you to do some final AMDInfo testing to confirm my new changes work for you too.

I've created a brand new AMDInfo v1.0.1, and it incorporates the logic we talked about earlier which will create the Eyefinity config in the AMD Radeon driver database if the user has removed it. It works for me, but I really want to make sure it works for you too.

So, can you please:
- Download the brand new AMDInfo v1.0.1 from https://github.com/terrymacdonald/AMDInfo/releases/tag/v1.0.1
- Delete the new-eyefinity-on.cfg and eyefinity-on.cfg config files
- Change to AMD Eyefinity using the AMD Radeon Setup
- Save a new eyefinity-really-on.cfg 
- Go into AMD Radeon Setup and delete the Eyfinity setup
- Change to dual-scree-back-main.cfg
- Change to eyefinity-really-on.cfg 

I'm hoping that discarding the Eyefinity setup will remove it from the driver database, and then that will test the new eyefinity creation code I've added to the library. If this works, then I'm considering the AMD Library done, and moving on to the next issues I need to fix for DM v2.0.0.

This final test will allow me to relax :).

Thanks
Terry

**Comment by scowling** [comment]
Log file attached.
Loading eyefinity from config *said *it worked OK.. but it didn't change
the display and AMD application has "Quick Setup" which I assume means it
believes there is nothing setup.

Let me know if you need a different log file

**Comment by terrymacdonald** [comment]
Darn it, no log file upon arrival here again.  I think the Github email server is stripping them off for some reason. Are you able to upload it through the Github GUI at https://github.com/terrymacdonald/DisplayMagician/issues/37 please?

**Comment by scowling** [comment]
Hmm, I thought I did. Oh well. This time I am _for sure_
[AMDInfo.log](https://github.com/terrymacdonald/DisplayMagician/files/7194255/AMDInfo.log)

**Comment by terrymacdonald** [comment]
LOL :).

Right. I have spent the last few hours digging down to the root of the problem, and I've now managed to craft a fix that works on my PC. It tries to use the existing SLSMap, and if it's not there anymore, it creates a new SLSMap that uses the same information, and then uses that. If you keep the AMD Radeon setup open while you run AMDInfo, you'll see the AMD Readeon setup update while AMDInfo runs. It's pretty cool.

Anyway - as before, I've created a brand new AMDInfo v1.0.2, and it *should* work. If it doesn't then I'm likely to have a little cry as I don't know what to do next :).

So, can you please:
- Download the brand, brand, brand new AMDInfo v1.0.2 from https://github.com/terrymacdonald/AMDInfo/releases/tag/v1.0.2
- Go into AMD Radeon Setup and delete the existing Eyefinity setup
- Change to dual-screen-back-main.cfg
- Change to eyefinity-really-on.cfg again

This time, instead of an error you *should* get Eyefinity turning on. It will take a good 20-30 seconds to do so, but it should eventually finish updating the screen. 

And can you please provide the AMDInfo.log file again whether it works or not. It will either confirm it works, or confirm my worst nightmare :D.

Thanks
Terry

**Comment by scowling** [comment]
First try, load eyefinity (using old eyefinity-really-on.cfg):
```
Program 'AMDInfo.exe' failed to run: Operation did not complete successfully because the file contains a virus or potentially 
 unwanted softwareAt line:1 char:1
+ .\AMDInfo.exe load .\eyefinity-really-on.cfg
```
No log file generated.

Second try, as administrator, load eyefinity (using old eyefinity-really-on.cfg): 
![image](https://user-images.githubusercontent.com/1783814/134075792-0f2d4bc6-4ed7-40a3-94e6-de49eb265bef.png)

Third, renabled eyefinity, tried to saved new config:
![image](https://user-images.githubusercontent.com/1783814/134076157-da9aac41-78f4-415c-b09b-9e8beb059acd.png)

No log file

**Comment by terrymacdonald** [comment]
Well that is odd...I'm an information security professional by day, and
none of my extensive security controls flagged the build as malicious on my
system. I also got VirusTotal to check the zip file as installed, and it
found nothing:
https://www.virustotal.com/gui/url/e7351f94ef0e3b77cbdfb612699d105d88013536a72ceabb717214734e003756

I'm trying to rack my brains as to why that would be happening, and I think
that AMDInfo.exe is being flagged by your AV as malicious for some reason,
and it is quarantining the file. Quarantining the file moves it to a
different location, so that would explain the two additional issues of it
wanting to open the config files, as it would have tried to find
AMDInfo.exe, but it wasn't there, so Windows instead thought you wanted to
load three different files, AMDInfo.exe, load, and the eyefinity.cfg. The
first two files didn't exist, so you only got the prompt for the config
file.

In my professional experience, these things happen sometimes when a string
of bytes in software just happens to match the same string in some other
malware family. Anti-virus (A/V) software often gets confused and alerts.
The weird thing is that VirusTotal tests files against over 50 different
A/V vendors, and AMDInfo v1.0.2.zip passed their tests just fine....

UPDATE: I just tried downloading this same file on another Win10 PC of mine
and I got the exact same error! I confirmed it's Windows Defender being
overly zealous:

[image: image.png]
Here is a link to another user who has something similar happen when they
made a python installer file:
https://www.reddit.com/r/learnpython/comments/ng3hmp/pyinstaller_create_onefile_exe_windows/

So TLDR is that I need to rebuild AMDInfo again later tonight, and then
make sure it won't trigger Windows Defender, and then once I do that I'll
update AMDInfo v1.0.2 and you can retry the test again.

Thanks

*Terry MacDonald *| Principal Security Consultant | Chief Operating Officer



M: +64 211 918 814 <+64+211+918+814>
E: ***@***.***
W: www.cosive.com

**Comment by scowling** [comment]
Yep, windows defender user here.
Let me know when you're ready with something.

**Comment by terrymacdonald** [comment]
@scowling So I ended up reporting it to Microsoft, who then went through and checked it, and they updated Windows Defender Antivirus so that it will exclude this file. See the submission info here: https://www.microsoft.com/en-us/wdsi/submission/d61c03ee-26ec-41e1-a415-8585385813ee

So, can you please update your Windows Defender A/V, and then try again? To update windows defender just do a windows update, and it will update to 1.349.1125.0 or later. I've tried to download the file myself now, and it now lets me with that 1.349.1125.0 A/V security intelligence update version.

Then, can you please:
- Download the brand, brand, brand new AMDInfo v1.0.2 from https://github.com/terrymacdonald/AMDInfo/releases/tag/v1.0.2
- Go into AMD Radeon Setup and delete the existing Eyefinity setup
- Change to dual-screen-back-main.cfg
- Change to eyefinity-really-on.cfg again

Eyefinity should turn on with that last command. It will take a good 20-30 seconds to do so, but it should eventually finish updating the screen. And as mentioned earlier can you please provide the AMDInfo.log file again whether it works or not. 

I have all my fingers and toes crossed willing this to work :)

Thanks
Terry

**Comment by scowling** [comment]
@terrymacdonald success.
I needed to create a new eyefinity setup and save it with the new version first though. The old cfg didn't work.

`AMDInfo v1.0.2
==============
By Terry MacDonald 2021

The display settings in .\dual-screen-front-main.cfg are already installed. No need to install them again. Exiting.

C:\Users\steve\Downloads\AMDInfo.v1.0.2 (1)> .\AMDInfo.exe load .\eyefinity-really-on.cfg

AMDInfo v1.0.2
==============
By Terry MacDonald 2021

Attempting to apply AMD display config from .\eyefinity-really-on.cfg
ERROR - AMDInfo AMD display settings were not applied correctly.

C:\Users\steve\Downloads\AMDInfo.v1.0.2 (1)> .\AMDInfo.exe save .\eyefinity-really-on2.cfg

AMDInfo v1.0.2
==============
By Terry MacDonald 2021

Display settings successfully saved to .\eyefinity-really-on2.cfg.

C:\Users\steve\Downloads\AMDInfo.v1.0.2 (1)> .\AMDInfo.exe load .\dual-screen-front-main.cfg

AMDInfo v1.0.2
==============
By Terry MacDonald 2021

Attempting to apply AMD display config from .\dual-screen-front-main.cfg
AMDInfo Display config successfully applied

C:\Users\steve\Downloads\AMDInfo.v1.0.2 (1)> .\AMDInfo.exe load .\eyefinity-really-on2.cfg

AMDInfo v1.0.2
==============
By Terry MacDonald 2021

Attempting to apply AMD display config from .\eyefinity-really-on2.cfg
AMDInfo Display config successfully applied

C:\Users\steve\Downloads\AMDInfo.v1.0.2 (1)>`

Log file attached
[AMDInfo.log](https://github.com/terrymacdonald/DisplayMagician/files/7200442/AMDInfo.log)

**Comment by terrymacdonald** [comment]
\o/     AWESOME!!!!!

OK - that's all I need from you now @scowling. You've singlehandedly helped me get AMD Eyefinity support into DisplayMagician, so just know that all the AMD simracers who use DisplayMagician v2.0.0 when it's released is directly down to your help. 

I'll be in touch when it's time to test the  next DisplayMagician v2.0.0 alpha.

Terry

**Comment by terrymacdonald** [comment]
I'll close this issue for now, as the code is ready to go when I release DisplayMagician v2.0.0. I revisit when I need help testing. This will help me keep track of the issues I actually need to work on before the final release.

You might like to see this @scowling 👍 

![image](https://user-images.githubusercontent.com/5721772/134146075-38ec4f68-d742-42e7-9883-1ddf3c9842f1.png)

**Comment by terrymacdonald** [comment]
@scowling Great news. I've got another DisplayMagican v2 build to test. This one has had a lot of local testing, and I've fixed a lot of bugs that I've found. And now I think it's ready for another round of testing. 

I'd really love it if you could test this out for me. To install DisplayMagician v2.0.0-alpha2:

- Firstly, uninstall DisplayMagician v2.0.0 alpha1 if you installed it earlier.
- Secondly, download and install DisplayMagician v2.0.0 alpha2 from this issue
- Thirdly, test it out :).

Hopefully it goes well for you and does everything it should, but if you do find any issues, please log them as new issues here: https://github.com/terrymacdonald/DisplayMagician/issues/new?assignees=terrymacdonald&labels=bug&template=bug_report.md&title=

Thanks 
Terry

[DisplayMagicianSetup-v2.0.0-alpha2.zip](https://github.com/terrymacdonald/DisplayMagician/files/7229458/DisplayMagicianSetup-v2.0.0-alpha2.zip)

**Comment by terrymacdonald** [comment]
And of course it supports AMD Eyefinity now thanks to you!

**Comment by terrymacdonald** [comment]
Great news! DisplayMagician v2.0.1 has been released, and it should fix this very issue. Please test it out, and if you still have any issues please log a new issue. You can get DisplayMagician v2.0.1 from here: https://github.com/terrymacdonald/DisplayMagician/releases/tag/v2.0.1

I'm closing this issue as the fix is in DisplayMagician v2.0.1, and I need to clear out the completed issues so I am able to work on the new issues that come in.

Thanks
Terry

**Comment by terrymacdonald** [comment]
ok @scowling, thanks to help from another user, I've found a couple of bugs that may have been causing issues for you too. Can you please download and install this new test build for me, and see how things go. If you still find some issues, then we'll have to do some testing with AMDInfo.

[DisplayMagicianSetup-v2.1.0.67.zip](https://github.com/terrymacdonald/DisplayMagician/files/7446035/DisplayMagicianSetup-v2.1.0.67.zip)

Thanks
Terry

**Comment by terrymacdonald** [comment]
Actually if you could now try and use the latest bugfix version to test, that would ensure you get the best, most likely to work version to test :)

Please install this new build, and do the same awesome testing you've been doing in the past! It's REALLY helping me out, so thank you. All the AMD users will owe you a beer (or other drink of your choice) 

[DisplayMagicianSetup-v2.1.0.71.zip](https://github.com/terrymacdonald/DisplayMagician/files/7447645/DisplayMagicianSetup-v2.1.0.71.zip)

Thanks
Terry

**Comment by terrymacdonald** [comment]
@scowling Looing at your logs, that NotifyIcon issue is one that I fixed recently. That issue was that games with really long names would break DisplayMagician. 

Anyway, can you please test this latest version instead of the others I posted above, and see if there this one works for you. I'd love to get all the issues fixed if I can before release :D.

[DisplayMagicianSetup-v2.1.0.197.zip](https://github.com/terrymacdonald/DisplayMagician/files/7491913/DisplayMagicianSetup-v2.1.0.197.zip)

Please report back when you can.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Hi @scowling,

We've recently found and fixed an error with the Windows Video Library that I made. It would come up with an error when it was validating a config, and that would stop the windows portion of the display config being applied. As mentioned this is now fixed.

Could you please update your DisplayMagician and test this version? [DisplayMagicianSetup-v2.1.0.215.zip](https://github.com/terrymacdonald/DisplayMagician/files/7531681/DisplayMagicianSetup-v2.1.0.215.zip)

IMPORTANT: I strongly recommend deleting all your existing Display Profiles and recreating them again. This will ensure that your Display Profiles have all the latest fixes applied to them. Once you recreate the Display Profiles you can then go through your shortcuts and update them to use the new profiles.

Thanks
Terry

**Comment by scowling** [comment]
![image](https://user-images.githubusercontent.com/1783814/141722059-5fd35e58-0b08-4527-84a4-b63de214944d.png)
Created a desk display profile and then a work shortcut
Was in the middle of creating a second profile and it was somehow linked to the work one.. so when I deleted the second one: this 
![image](https://user-images.githubusercontent.com/1783814/141722929-86677554-49e2-467b-b25e-4e2b3987e6f6.png)

**Comment by terrymacdonald** [comment]
Thanks Steve,

Yeah it turns out I added a new bug with the last UI changes I made. Sigh.

I'll update DM to fix that, and add a new copy shortcut function usability
improvement. I have another bug for NVIDIA to fix, but as that doesn't
affect you, I might get you to test and earlier build than the NVIDIA folk.
Hope that's ok!

Thanks
Terry

**Comment by scowling** [comment]
That's fine; I recently updated my GPU but am still AMD so happy to help where I can

**Comment by terrymacdonald** [comment]
OK - Fixed! Had to change way more than I thought I would, but it the UI *should* work now. Can you please test DisplayMagician v2.1.0.239 and see what happens. 

[DisplayMagicianSetup-v2.1.0.239.zip](https://github.com/terrymacdonald/DisplayMagician/files/7537310/DisplayMagicianSetup-v2.1.0.239.zip)

I am very interested in whether you can successfully swap from a triple screen profile to an eyefinity screen, and back again successfully. And if that works the first time, then whether you can do the same thing after 1 or 2 reboots of your PC. If that all works, then I'm going to be so happy!

Thanks for helping all the AMD users @scowling! You rock.
Terry

**Comment by terrymacdonald** [comment]
Hi @scowling,

So domenic and I found some errors in the Windows CCD WinLibrary which would affect you as an AMD user (see #41). We worked over the last week and a half to get the bugs squashed, and I've released them in a new test build of DisplayMagician here: [DisplayMagicianSetup-v2.1.0.254.zip](https://github.com/terrymacdonald/DisplayMagician/files/7569245/DisplayMagicianSetup-v2.1.0.254.zip)

This is very nearly production ready as I've run out of any more bugs to squash. So can you please test this pre-release version to make sure it's fit for release?

**IMPORTANT: If you have moved to WIndows 11, please delete all your DisplayProfiles as you need to recreate them if you moved to Windows 11 (each DisplayProfile is specific to the operating system it was made on). Then change all your game shortcuts to use the new DisplayProfiles.**

I'm hoping that once you've recreated your DisplayProfiles and updated your Game Shortcuts then everything will work as it should 🤞 

Good luck!

Thanks
Terry

**Comment by scowling** [comment]
Seems to be OK from what limited testing I have had time to do

Steve

**Comment by terrymacdonald** [comment]
That's great news. If it does Eyefinity then we're all good :)

Thanks
*Terry *

**Comment by terrymacdonald** [comment]
Hi @scowling,

I've just finished DisplayMagician v2.3.0.35 which is hopefully the next big release of DisplayMagician. It has a lot of improvements and bug fixes to the NVIDIALibrary, but it also has a few improvements to the AMDLibrary too. I don't have a supported AMD video card to test with unfortunately, so I was hoping you would be able to do some testing for me.

[DisplayMagicianSetup-v2.3.0.35.zip](https://github.com/terrymacdonald/DisplayMagician/files/8457813/DisplayMagicianSetup-v2.3.0.35.zip)

Can you please download and install DisplayMagician v2.3.0.35, and then do the following steps:

- Delete your old Display Profiles 
- Create new Display Profiles 
- Update your Game Shortcuts to use the new Display Profiles
- Test out Eyefinity, cloned displays, and see if anything breaks

If you have any issues, please create a Support ZIP File (main screen > Settings > Create Support ZIP File) and upload it here through your web browser. I'll then be able to use that for any bug fixing.
 
Thanks
Terry

**Comment by scowling** [comment]
Hey @terrymacdonald 
Absolutely swamped with work at the moment; unlikely to be able to test until after Easter .. that ok?

**Comment by terrymacdonald** [comment]
Yeah of course! There's no pressure on the timeline. Just whenever is fine.
Youcve just got such a unique setup that I know if it works for you or will
work for anyone... That's the only reason I ask 🙂.

Feel free to tell me to stop contacting you if you'd prefer. I don't want
you to feel like you have to help all the time. So seriously please tell me
when you like to stop testing.

Thanks
Terry

**Comment by terrymacdonald** [comment]
Sorry! We just found an issue with some botched Display Profile upgrade logic :(.  Now fixed.

Please do the following:

- Delete `C:\Users\<Username>\AppData\Local\Displaymagician\Profiles\DisplayProfiles_2.3.json` if it exists.
- Install this new DisplayMagician version:
[DisplayMagicianSetup-v2.3.0.36.zip](https://github.com/terrymacdonald/DisplayMagician/files/8467396/DisplayMagicianSetup-v2.3.0.36.zip)
- Run DisplayMagician
- Create your Display Profiles again the way you want them
- Update your Game Shortcuts to use the new Display Profiles
- Have fun with your games!

Hopefully it should work this time.  Please log a new issue at https://github.com/terrymacdonald/DisplayMagician/issues/new/choose if you experience any new bugs that you've not seen before, and I'll get right on fixing them.

Thanks
Terry

**Comment by terrymacdonald** [comment]
@scowling I have recently added code so that DisplayMagician knows the rotation of screens on an AMD card, and shows the rotation properly in the UI. I do not have an AMD card, so I need one of my AMD users to test that it works before I am able to release the updated DisplayMagician version to everyone.

I have created a test version of DisplayMagician and have linked to it below. Can you please download, unzip and install this version to make sure it works? Please report back here in this issue so I know even if it works fine, so I know.

[DisplayMagicianSetup-v2.5.0.337.zip](https://github.com/terrymacdonald/DisplayMagician/files/10312741/DisplayMagicianSetup-v2.5.0.337.zip)

Thanks
Terry

**Comment by scowling** [comment]
Installs OK and loads my existing profiles ok.
Anything else you need me to confirm?

**Comment by terrymacdonald** [comment]
Hi Steve,

Thanks for doing this. I really appreciate your help!

Can you please rotate one of your screens in windows, then see if that
screen rotation is shown in DisplayMagician in the display profiles window.
Can you please take a screenshot of it and drop it in this issue?

Also can you please save a new profile with the rotated screen, same then
see if you can swap back and forth from your existing display profiles to
the rotated screen profile.

This will confirm that it all works sad intended!

Thanks
Terry

**Comment by scowling** [comment]
Ugh. I found some time and attempted to play around with AMD settings but have hit this issue: https://www.amd.com/en/support/kb/faq/pa-300
The solution they propose is to turn off ALL windows updates for devices; which I'm not really prepared to do.

I will have to investigate and come back to this.

**Comment by terrymacdonald** [comment]
Oh no! Really sorry to hear that. Take whatever time you need. I don't want
you to break your setup on my account.

I have asked a few others to test it the AMD side, so feel free to not do
any more testing for this. Completely happy as I'm sure others will come
back of there is a problem.

Thanks
Terry
